contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1476,22 +1516,37 @@ handleForJsonAuto(Query *query)
 				foreach(lc, subq->targetList) {
 					TargetEntry* te = castNode(TargetEntry, lfirst(lc));
 					int oid = te->resorigtbl;
+					String* s = castNode(String, lfirst(list_nth_cell(colnameAlias->colnames, i)));
```

Comment:
This code is appearing in two different places, should write into the same function to avoid repeating itself.